File: covid-api/src/main/java/com/tacs/grupo1/covid/api/dto/CountryList.java
package com.tacs.grupo1.covid.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class CountryList {
    private Long id;
    private String name;
    private Set<Country> countries;
    @JsonProperty("user_id")
    private Long userId;
    @JsonProperty("creation_date")
    private String creationDate;
}

File: covid-frontend/src/models/CountryList.ts
import Country from "./Country";

export default class CountryList{
    public id: number;
    public name: string;
    public countries: Array<Country>;
    public user_id: number;
    public creation_date: String;

    constructor(countryList?: any){
        if(countryList){
            this.id = countryList.id;
            this.name = countryList.name;
            this.countries = countryList.countries.map((country: any) => {return new Country(country)});
            this.user_id = countryList.user_id;
            this.creation_date = countryList.creation_date;
        }else{
            this.id = 0;
            this.name = "";
            this.countries = [];
            this.user_id = 0;
            this.creation_date = "";
        }
    }
}